Alternate lines

lbud opened the pull request:
This PR adds an option to `L.Routing.Control` to allow for displaying alternate routes as polylines where available, where `showAlternatives` is a boolean (defaults to `false`) and `altLineOptions` mirrors `lineOptions` to style alternate line(s). Alternative lines are clickable to switch focus to that route (which also changes the itinerary display to indicate that the clicked line is chosen). (We want to use this in the osrm-frontend rewrite.) In practice that looks like this:

![lrm-altlineoptions](https://cloud.githubusercontent.com/assets/3170312/9459538/7ea26ce4-4ab1-11e5-8d47-a134f752874b.gif)

@perliedman — if this looks good to you I'll go ahead and add any necessary docs/tests.

perliedman commented on src/L.Routing.Itinerary.js:
```diff
@@ -107,6 +107,10 @@
 			this[collapsed ? 'show' : 'hide']();
 		},
 
+		_dispatchRoutes: function(e) {
```
Should this method be moved to `L.Routing.Control`? It looks like it's only used there.

perliedman commented:
Wow, thanks! This looks great. :+1: Both a generally very nice idea from a UX perspective, and looks like a solid/straight forward implementation.

I added a minor comment about some code structure above.

Also, I don't think I understand the difference between the existing `routeselected` event and the new `dispatchroutes` event - could you perhaps elaborate a bit on why the new event is needed?

lbud commented:
@perliedman yeah, I was trying to remember — I think I had it there because it was doing something else before, but it's definitely not needed. Removed and switched to the existing `routeselected` event. I also added some options to the `examples/index.js` file (not sure whether you'd want to use it in the example or not) and I forgot to mention but you probably saw there's one unrelated bugfix in this, which passes booleans as strings to be included in the URL hash.

Other than that, this should be good to go — adding docs PR to gh-pages branch.

perliedman commented:
Thanks again for this, a really great addition!
